ios/FluentUI/Date Time Pickers/Date Picker/DatePickerController.swift

```diff
@@ -93,13 +95,14 @@ class DatePickerController: UIViewController, GenericDateTimePicker {
     /// - Parameters:
     ///   - startDate: A date object for the start day or day/time to be initially selected.
     ///   - endDate: A date object for an end day or day/time to be initially selected.
+    ///   - calendarConfiguration: The configuration for the picker to override the default first weekday, reference start date, and reference end date
     ///   - datePickerMode: The MSDateTimePicker mode this is presented in.
     ///   - selectionMode: The side (start or end) of the current range to be selected on this picker.
     ///   - rangePresentation: The `DateRangePresentation` in which this controller is being presented if `mode` is `.dateRange` or `.dateTimeRange`.
     ///   - titles: A `Titles` object that holds strings for use in overriding the default picker title, subtitle, and tab titles. If title is not provided, titleview will show currently selected date. If tab titles are not provided, they will default to "Start Date" and "End Date".
     ///   - leftBarButtonItem: optional UIBarButtonItem to be presented as left bar-button.
     ///   - rightBarButtonItem: optional UIBarButtonItem to be presented oas right bar-button. Note that if this view is presented, the Confirm button is not generated automatically.
-    init(startDate: Date, endDate: Date, mode: DateTimePickerMode, selectionMode: DatePickerSelectionManager.SelectionMode = .start, rangePresentation: DateTimePicker.DateRangePresentation, titles: DateTimePicker.Titles?, leftBarButtonItem: UIBarButtonItem?, rightBarButtonItem: UIBarButtonItem?) {
+    init(startDate: Date, endDate: Date, calendarConfiguration: CalendarConfiguration? = nil, mode: DateTimePickerMode, selectionMode: DatePickerSelectionManager.SelectionMode = .start, rangePresentation: DateTimePicker.DateRangePresentation, titles: DateTimePicker.Titles?, leftBarButtonItem: UIBarButtonItem?, rightBarButtonItem: UIBarButtonItem?) {
```

Comment:
Yes that's definitely an option. I have this "calendarConfiguration: CalendarConfiguration? = nil" in a lot of places. Do we want to change ALL of them to calendarConfiguration: CalendarConfiguration? = CalendarConfiguration.default? Or maybe just one of the ones further up the stack can be the default, and all the calls further down the stack can just use "calendarConfiguration: CalendarConfiguration". Not sure what best practice is here

Reply:
All of them was my suggestion, but feel free to disagree if you think it makes sense to keep it as is

Reply:
Fine with all three (all nil, or all default, or some default/some required parameters). Are there any advantages to switching them from all nil to all default? 

I think a possible benefit of keeping it as nil until the very end is that we can differentiate between whether the creator of the date picker decided not to pass in anything at all, or actually passed in the default value. This could be useful for debugging, or maybe for new features in the future (although I can also see an argument that these two scenarios should always be treated the same way). 

Not a strong opinion though,

Reply:
My reasoning was I mostly try to avoid the keyword "nil" because null pointer exceptions scare me, and it felt like the default value of an optional value should be.. the default configuration.

I think those two aren't strong reasons though, and I couldn't really find any guidance on aka.ms/swift-guide or docs.swift.org/swift-book/LanguageGuide/Functions.html saying otherwise. As such, I think this is mostly a stylistic choice, so I'll leave it up to you what you prefer.

Reply:
I think that if all the instances we'd be using the calendar configuration are something like below, it makes sense to make the parameter type non-optional and just provide the CalendarConfiguration.default as the default parameter.

`calendarConfiguration?.referenceStartDate ?? CalendarConfiguration.default.referenceStartDate`

It'd be just simpler because it'd avoid these checks everywhere same code above would be replaced by just:

`calendarConfiguration.referenceStartDate`

Reply:
Okay just pushed a commit that replaces a bunch of places where I had `calendarConfiguration: CalendarConfiguration? = nil` with just `calendarConfiguration: CalendarConfiguration`. I realized while doing this that  there were two variables startYear and endYear in `case date(startYear: Int, endYear: Int)` in the DateTimePickerViewMode enum that I think are redundant now after introducing calendarConfiguration. The code is kind of confusing but as far as I can tell, before this PR these variables always ended up as the default calendar configuration. I also went ahead and removed static let defaultStartYear and defaultEndYear in DateTimePickerViewMode enum. 

The two changes in this commit (making calendarConfiguration parameters non-optional/no default) and the changes to the DateTimePickerViewMode enum are related. If the above change to the DateTimePickerViewMode enum seems out of scope for this PR, I can revert this whole commit.